nixos/vimPlugins: add plugins + update

gjabell opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
Missing plugins in vimPlugins.

###### Things done
Update vimPlugins; add the following vimPlugins:
- calendarVim
- gv-vim
- kotlin-vim
- vim-salve
- utl-vim
- vim-clojure-highlight
- vim-clojure-static
- vim-sexp
- vim-sexp-mappings-for-regular-people
- VimOrganizer

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

gjabell committed:
vimPlugins: Update

gjabell committed:
vimPlugins.mattn-calendar-vim: init at 2018-08-25

gjabell committed:
vimPlugins.gv-vim: init at 2019-10-13

gjabell committed:
vimPlugins.kotlin-vim: init at 2019-05-26

gjabell committed:
vimPlugins.vim-salve: init at 2019-08-02

gjabell committed:
vimPlugins.utl-vim: init at 2010-10-18

gjabell committed:
vimPlugins.vim-clojure-highlight: init 2015-07-05

gjabell committed:
vimPlugins.vim-clojure-static: init at 2017-10-23

gjabell committed:
vimPlugins.vim-sexp: init at 2017-05-15

gjabell committed:
vimPlugins.vim-sexp-mappings-for-regular-people:

init at 2014-11-04

gjabell committed:
vimPlugins.VimOrganizer: init at 2014-04-10

timokau commented on pkgs/misc/vim-plugins/overrides.nix:
```diff
@@ -91,6 +91,18 @@ self: super: {
     };
   };
 
+  # do not auto-update this one, as the name clashes with calendar-vim
```
In that case we should have some way for the user to distinguish the two. Maybe it was a mistake to have the plugins in nixpkgs without namespaces.

Ideally you'd implement alias support in our plugin update infrastructure (`calendar-vim as mattn-calendar-vim` in `plugin-list`), but if you don't want to do that we should at least choose a more descriptive name.

gjabell replied:
I'm fine with adding that as an alias in `plugin-list`; should the package definition stay where it is in `overrides.nix`? If it's added to `vim-plugin-list` I'm assuming it'll have a naming conflict with `calendar-vim`.

timokau replied:
No I meant that if you want to go down that road, you'd have to adapt `update.py` to work with statements like `calendar-vim as mattn-calendar-vim` in `plugin-list`. That way no override would be necessary and auto updates would work.

gjabell replied:
Ah, gotcha. I can look into that, seems like it would be a useful feature to have. Would it be best to put this PR on hold and merge that in separately?

timokau replied:
Whatever way you prefer. As long as the change to `update.py` is in a separate commit that is fine by me. If you want to do a manual override as a stopgap first that is also okay.

gjabell replied:
Great, sounds good to me.

timokau replied:
Okay, ping me if you do decide to open a different PR for it.

gjabell replied:
@timokau Just pushed changes to `update.py` to add alias support; see the first commit in the PR. Also reupdated plugins in the update commit, and moved `vim-docbk-snippets` from `overrides.nix` into `vim-plugin-names` since it works with the alias support :)

timokau reviewed (changes requested):
LGTM except the naming issue.

gjabell committed:
vimPlugins: Add alias support to update.py

Plugins in `plugin-list` can now optionally specify an alias name, to
avoid naming collisions if plugins have the same repo name. For example,
specifying `author/common-plugin as author-common-plugin` will use
`author-common-plugin` as the package name in `generated.nix`.

gjabell committed:
vimPlugins: fix vim-docbk-snippets naming clash

timokau reviewed (approved):
Looks good to me, thank you!

I'd like at least one more person with some nix-vim experience to review this. @teto / @Ma27 ?

jonringer commented:
we should also update `doc/languages-frameworks/vim.section.md` to mention aliasing as construct. otherwise diff LGTM

jonringer commented:
but that can be done in another PR, I'll get this in as it's been a while

jonringer reviewed (approved):
`nix-review` passes on NixOS
diff LGTM
commits LGTM

```
[98 built, 33 copied (881.5 MiB), 204.9 MiB DL]
https://github.com/NixOS/nixpkgs/pull/72241
49 package were build:
vimPlugins.VimOrganizer vimPlugins.alchemist-vim vimPlugins.awesome-vim-colorschemes vimPlugins.calendar-vim vimPlugins.coc-eslint vimPlugins.coc-imselect vimPlugins.coc-java vimPlugins.coc-tsserver vimPlugins.editorconfig-vim vimPlugins.falcon vimPlugins.fzf-vim vimPlugins.ghcid vimPlugins.gv-vim vimPlugins.kotlin-vim vimPlugins.lightline-vim vimPlugins.mattn-calendar-vim vimPlugins.ncm2-ultisnips vimPlugins.neosnippet-vim vimPlugins.ultisnips vimPlugins.unicode-vim vimPlugins.utl-vim vimPlugins.vim-airline vimPlugins.vim-airline-themes vimPlugins.vim-android vimPlugins.vim-clojure-highlight vimPlugins.vim-clojure-static vimPlugins.vim-devicons vimPlugins.vim-dirvish vimPlugins.vim-docbk-snippets vimPlugins.vim-elixir vimPlugins.vim-glsl vimPlugins.vim-go vimPlugins.vim-jsx-pretty vimPlugins.vim-ledger vimPlugins.vim-plug vimPlugins.vim-polyglot vimPlugins.vim-projectionist vimPlugins.vim-salve vimPlugins.vim-sexp vimPlugins.vim-sexp-mappings-for-regular-people vimPlugins.vim-signify vimPlugins.vim-slime vimPlugins.vim-sneak vimPlugins.vim-snippets vimPlugins.vim-terraform vimPlugins.vim-test vimPlugins.vimtex vimPlugins.yats-vim vimPlugins.youcompleteme
```
